### PR TITLE
Add RRTMGPModel and radiation cache/tendency/callback

### PR DESCRIPTION
**Pull request by dennisYatunin**
This PR adds two important files---`rrtmgp_model.jl` and `radiation_utilities.jl`.

`rrtmgp_model.jl`is a user-friendly wrapper for `RRTMGP.jl` that is able to automatically handle extension profiles, boundary layers, and interpolation/extrapolation schemes, as well as expose all the relevant inputs/outputs of `RRTMGP` to users while hiding all  the irrelevant data structures. We will eventually move this file into the `RRTMGP.jl` repository, but, for the sake of faster development, we should keep it here until we have settled on the final interface.

`radiation_utilities.jl` is the equivalent of `baroclinic_wave_utilities.jl` for radiation; it introduces a cache function, a tendency function, and a callback function that use an `RRTMGPModel`. A lot of information is currently hardcoded into these functions (e.g., volume mixing ratios of gases are read in from a dataset provided with `RRTMGP`, and the land surface temperature is set to `29 * exp(-(ϕ / 26)^2 / 2) + 271`), but we will add more versatility as we continue to experiment.

This PR also adds a new `baroclinic_wave_rhoe_equilmoist_radiation.jl` example to the Buildkite pipeline. This example demonstrates how to use `radiation_utilities.jl`. Since it is not very physically meaningful, this example will eventually be removed, so a regression test does not need to be added for it.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -185,6 +185,10 @@ steps:
         command: "julia --color=yes --project=examples examples/hybrid/driver.jl --TEST_NAME baroclinic_wave_rhoe_equilmoist --job_id sphere_baroclinic_wave_rhoe_equilmoist --regression_test true"
         artifact_paths: "sphere_baroclinic_wave_rhoe_equilmoist/*"
 
+      - label: ":computer: baroclinic wave (ρe) equilmoist radiation"
+        command: "julia --color=yes --project=examples examples/hybrid/driver.jl --TEST_NAME sphere/baroclinic_wave_rhoe_equilmoist_radiation --job_id sphere_baroclinic_wave_rhoe_equilmoist_radiation"
```
Comment (charleskawczynski) [inline .buildkite/pipeline.yml:189]:
```suggestion
        command: "julia --color=yes --project=examples examples/hybrid/driver.jl --TEST_NAME baroclinic_wave_rhoe_equilmoist_radiation --job_id sphere_baroclinic_wave_rhoe_equilmoist_radiation"
```

Comment (dennisYatunin) [reply .buildkite/pipeline.yml:189]: Thank you!

**Review by szy21 — APPROVED** [review]

File: examples/hybrid/sphere/baroclinic_wave_rhoe_equilmoist_radiation.jl [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# Variables required for driver.jl (modify as needed)
+params = BaroclinicWaveParameterSet()
+horizontal_mesh = baroclinic_wave_mesh(; params, h_elem = 4)
+npoly = 4
+z_max = FT(30e3)
+z_elem = 10
+dt_save_to_disk = FT(0) # 0 means don't save to disk
+ode_algorithm = OrdinaryDiffEq.Rosenbrock23
+jacobian_flags = (; ∂ᶜ𝔼ₜ∂ᶠ𝕄_mode = :no_∂ᶜp∂ᶜK, ∂ᶠ𝕄ₜ∂ᶜρ_mode = :exact)
```
Comment (charleskawczynski) [inline examples/hybrid/sphere/baroclinic_wave_rhoe_equilmoist_radiation.jl:18]: I think we can delete some of this, as it's been moved into `driver.jl`

File: examples/hybrid/sphere/baroclinic_wave_rhoe_equilmoist_radiation.jl [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+function postprocessing(sol, output_dir)
+    @info "L₂ norm of ρe at t = $(sol.t[1]): $(norm(sol.u[1].c.ρe))"
+    @info "L₂ norm of ρe at t = $(sol.t[end]): $(norm(sol.u[end].c.ρe))"
+
+    anim = Plots.@animate for Y in sol.u
+        ᶜv = Geometry.UVVector.(Y.c.uₕ).components.data.:2
+        Plots.plot(ᶜv, level = 3, clim = (-6, 6))
+    end
+    Plots.mp4(anim, joinpath(output_dir, "v.mp4"), fps = 5)
+end
```
Comment (charleskawczynski) [inline examples/hybrid/sphere/baroclinic_wave_rhoe_equilmoist_radiation.jl:54]:
```suggestion
```
I think a generic form of this was moved into (I think the name is) `define_post_processing.jl`

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -185,6 +185,10 @@ steps:
         command: "julia --color=yes --project=examples examples/hybrid/driver.jl --TEST_NAME baroclinic_wave_rhoe_equilmoist --job_id sphere_baroclinic_wave_rhoe_equilmoist --regression_test true"
         artifact_paths: "sphere_baroclinic_wave_rhoe_equilmoist/*"
 
+      - label: ":computer: baroclinic wave (ρe) equilmoist radiation"
+        command: "julia --color=yes --project=examples examples/hybrid/driver.jl --TEST_NAME baroclinic_wave_rhoe_equilmoist_radiation --job_id sphere_baroclinic_wave_rhoe_equilmoist_radiation"
+        artifact_paths: "sphere_baroclinic_wave_rhoe_equilmoist/*"
```
Comment (szy21) [inline .buildkite/pipeline.yml:190]:
```suggestion
        artifact_paths: "sphere_baroclinic_wave_rhoe_equilmoist_radiation/*"
```

**Pull request merged**